extensions/community/CrazyGamesAdApi.json

```
@@ -0,0 +1,393 @@
+          "actions": [
+            {
+              "type": {
+                "value": "SetReturnBoolean"
+              },
+              "parameters": [
+                ""
+              ]
+            }
+          ]
+        },
+        {
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [
+            {
+              "type": {
+                "value": "SceneVariableAsBoolean"
+              },
+              "parameters": [
+                "__CrazyGamesExtension.adFinished",
+                "True"
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "type": {
+                "value": "SetReturnBoolean"
+              },
+              "parameters": [
+                "True"
+              ]
+            }
+          ]
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Check if Ad had an error.",
+      "fullName": "Check if Ad had an error",
+      "functionType": "Condition",
+      "group": "Crazy Games",
+      "name": "AdHadAnError",
+      "sentence": "Ad had an error?",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [],
+          "actions": [
+            {
+              "type": {
+                "value": "SetReturnBoolean"
+              },
+              "parameters": [
+                ""
+              ]
+            }
+          ]
+        },
+        {
+          "type": "BuiltinCommonInstructions::Standard",
+          "conditions": [
+            {
+              "type": {
+                "value": "SceneVariableAsBoolean"
+              },
+              "parameters": [
+                "__CrazyGamesExtension.adError",
+                "True"
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "type": {
+                "value": "SetReturnBoolean"
+              },
+              "parameters": [
+                "True"
+              ]
+            }
+          ]
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Returns an invite link.",
+      "fullName": "Returns an invite link",
+      "functionType": "ExpressionAndCondition",
+      "name": "InviteLink",
+      "sentence": "Returns an invite link _PARAM1_",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "try {",
+            "    const crazysdk = window.CrazyGames.CrazySDK.getInstance();",
+            "    crazysdk.init();",
+            "",
+            "    const roomId = eventsFunctionContext.getArgument(\"RoomId\");",
+            "    const linkToShare = crazysdk.inviteLink({ roomId: roomId });",
+            "",
+            "    eventsFunctionContext.returnValue = linkToShare;",
+            "} catch(e) {",
+            "    console.log({e});",
```

Always include a full error message. Also avoid to name your variables with a single letter, it's hard to read and hard to rename.
```suggestion
            "    console.log("Unable to create an invite link. Full error is:", error);",
```

---

updated